authentication-wrapper/src/main/java/io/mosip/idp/authwrapper/service/IdentityAuthenticationService.java

```diff
@@ -5,35 +5,261 @@
+                    .body(requestBody);
+            requestEntity.getHeaders().add(SIGNATURE_HEADER_NAME, getRequestSignature(requestBody));
+            IdaSendOtpResponse idaSendOtpResponse = restTemplate.exchange(requestEntity, IdaSendOtpResponse.class).getBody();
+            if(idaSendOtpResponse != null && CollectionUtils.isEmpty(idaSendOtpResponse.getErrors()) &&
+                    idaSendOtpResponse.getResponse() != null) {
+                return new SendOtpResult(idaSendOtpResponse.getTransactionID(), idaSendOtpResponse.getMaskedEmail(),
+                        idaSendOtpResponse.getMaskedMobile());
+            }
+            log.error("Errors in response received from IDA : {}", idaSendOtpResponse.getErrors());
+            throw new SendOtpException(idaSendOtpResponse.getErrors().get(0).getErrorCode());
+        } catch (Exception e) {
+            log.error("send-otp failed with clientId : {}", clientId, e);
+        }
+        throw new SendOtpException(REQUEST_FAILED);
+    }
+
+    private void buildAuthRequest(String authFactor, String authChallenge,
+                                  IdaKycAuthRequest.AuthRequest authRequest) {
+        log.info("Build kyc-auth request with authFactor : {}",  authFactor);
+        switch (authFactor.toUpperCase()) {
+            case "OTP" : authRequest.setOtp(authChallenge);
+                break;
+            case "PIN" : authRequest.setStaticPin(authChallenge);
+                break;
+            case "BIO" :
+                byte[] decodedBio = IdentityProviderUtil.B64Decode(authChallenge);
+                try {
+                    List<IdaKycAuthRequest.Biometric> biometrics = objectMapper.readValue(decodedBio,
+                            new TypeReference<List<IdaKycAuthRequest.Biometric>>(){});
+                    authRequest.setBiometrics(biometrics);
+                } catch (IOException e) {
+                    log.error("Failed to parse biometric capture response", e);
+                }
+                break;
+            default:
+                throw new NotImplementedException("KYC auth not implemented");
+        }
+    }
+
+    private byte[] getCertificateThumbprint(Certificate certificate) {
+        try {
+            return DigestUtils.sha256(certificate.getEncoded());
+        } catch (CertificateEncodingException e) {
+            log.error("Failed to get cert thumbprint", e);
+        }
+        return new byte[]{};
+    }
+
+    private String getRequestSignature(String request) {
+        JWTSignatureRequestDto jwtSignatureRequestDto = new JWTSignatureRequestDto();
+        jwtSignatureRequestDto.setApplicationId(Constants.IDP_PARTNER_APP_ID);
+        jwtSignatureRequestDto.setReferenceId("");
+        jwtSignatureRequestDto.setIncludePayload(false);
+        jwtSignatureRequestDto.setIncludeCertificate(false);
+        jwtSignatureRequestDto.setDataToSign(IdentityProviderUtil.B64Encode(request));
```

Comment:
I see some places we use crypto util base64 and some places IdentityProviderUtil.B64Encode, any diffference between the two ?

Reply:
Both are same vishwanath.

Reply:
If both are same, can we agree on one method and use the same everywhere in idp ?

Reply:
Now throughout we are only using IdentityProviderUtil.b64Encode() method.